src/compute/src/rpc/service/stream_service.rs

```diff
@@ -138,6 +138,24 @@ impl StreamService for StreamServiceImpl {
         let barrier =
             Barrier::from_protobuf(req.get_barrier().unwrap()).map_err(StreamError::from)?;
 
+        // The barrier might be outdated and been injected after recovery in some certain extreme
+        // scenarios. So some newly creating actors in the barrier are possibly not rebuilt during
+        // recovery. Check it here and return an error here if some actors are not found to
+        // avoid collection hang. We need some refine in meta side to remove this workaround since
+        // it will cause another round of unnecessary recovery.
+        let actor_ids = self.mgr.all_actor_ids().await;
+        if req
+            .actor_ids_to_collect
+            .iter()
+            .any(|id| !actor_ids.contains(id))
+        {
```

Comment:
How about also recording the problematic actor ID in the log and error message?

```suggestion
        if let Some(id) = req
            .actor_ids_to_collect
            .iter()
            .filter(|id| !actor_ids.contains(id))
            .next()
        {
```

Reply:
Sure, I changed it to record all missing actor ids in the log. 😄